src/Tools/dotnet-monitor/Egress/AzureBlob/AzureBlobEgressProvider.cs

```diff
@@ -116,8 +124,23 @@ public AzureBlobEgressProvider(ILogger<AzureBlobEgressProvider> logger)
                 // Write blob headers
                 await blobClient.SetHttpHeadersAsync(CreateHttpHeaders(artifactSettings), cancellationToken: token);
 
-                // Write blob metadata
-                await blobClient.SetMetadataAsync(artifactSettings.Metadata, cancellationToken: token);
+                try
+                {
+                    Dictionary<string, string> mergedMetadata = artifactSettings.Metadata.Union(artifactSettings.CustomMetadata).ToDictionary(pair => pair.Key, pair => pair.Value);
+
+                    // Write blob metadata
+                    await blobClient.SetMetadataAsync(mergedMetadata, cancellationToken: token);
+                }
+                catch (Exception ex) when (ex is InvalidOperationException || ex is RequestFailedException)
+                {
+                    Logger.InvalidMetadata(ex);
+                    await blobClient.SetMetadataAsync(artifactSettings.Metadata, cancellationToken: token);
+                }
+                catch (ArgumentException ex)
```

Comment:
Let's not handling this exception here because it masks when other parts of the code throw ArgumentException.